core/include/sched.h

```diff
@@ -188,6 +188,25 @@ void sched_register_cb(void (*callback)(uint32_t, uint32_t));
 
 #endif
 
+#ifdef BOARD_NATIVE
```

Comment:
yes, a separate header with the compatibility additions is highly preferable :+1: .